router/cmd/custom/module/module.go

```diff
@@ -104,6 +111,7 @@ func (m *MyModule) Module() core.ModuleInfo {
 // Interface guard
 var (
 	_ core.RouterMiddlewareHandler = (*MyModule)(nil)
+	_ core.PreRequestMiddleware    = (*MyModule)(nil)
```

Comment:
I'd like to suggest a different name. I think `RouterOnRequest` is more appropriate.

Reply:
Are you suggesting `RouterOnRequestMiddlewareHandler` or `RouterOnRequest`?

Separate note, actually missed adding "Handler" originally

Reply:
I have a worry that people might get confused between `RouterMiddleware` and `RouterOnRequest`, do you think `RouterOnRequestStart` might make more sense?

Reply:
I like your idea but I don't believe it becomes an issue. Also if you add a `Start` people are looking for `End`. I think this creates inconsistency. Look at https://fastify.dev/docs/v2.15.x/Documentation/Lifecycle/ it reflects the proposed structure.

Reply:
I don't have a problem with the name itself, more about a worry of a distinction with the "RouterMiddleware" hook (Fastify does not seem to have that problem since they don't have 'RouterMiddleware"). but agree on folks looking for the `End` if we do `Start`.

However I checked some other frameworks as well and a few seem to also use `OnRequest` also, so will stick to that.

Reply:
Modified